ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala

```diff
@@ -4,72 +4,101 @@
 package com.daml.ledger.participant.state.kvutils
 
 import java.io.{ByteArrayInputStream, ByteArrayOutputStream, DataInputStream, DataOutputStream}
-
 import com.daml.ledger.offset.Offset
 
+import java.nio.ByteBuffer
+
 /** Helper functions for generating 16 byte [[Offset]]s from integers.
   * The created offset will look as follows:
-  * | highest index (64 bits) | middle index (32 bits) | lowest index (32 bits) |
+  * | version (8 bits) | highest index (56 bits) | middle index (32 bits) | lowest index (32 bits) |
   * Leading zeros will be retained when generating the resulting offset bytes.
   *
   * Example usage:
   *
-  *   - If you have one record per block then use [[OffsetBuilder.fromLong]] with a single argument, the block ID.
-  *   - If you may have multiple records per block then use [[OffsetBuilder.fromLong]] with the index within the block as the second argument.
+  *   - If you have one record per block then use [[OffsetBuilder.of]] with a single argument, the block ID.
+  *   - If you may have multiple records per block then use [[OffsetBuilder.of]] with the index within the block as the second argument.
   *
   * @see com.daml.ledger.offset.Offset
   * @see com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantStateReader
   */
 object OffsetBuilder {
-  private[kvutils] val highestStart = 0
+  val DefaultVersion: Byte = 0
+  val HighestIndexSize = 7
+
+  private[kvutils] val highestStart = 1
   private[kvutils] val middleStart = 8
   private[kvutils] val lowestStart = 12
   private[kvutils] val end = 16
 
   def onlyKeepHighestIndex(offset: Offset): Offset = {
```

Comment:
I think you're going to need a method to extract the version, too.

Reply:
Isn't this one enough: https://github.com/digital-asset/daml/pull/10668/files#diff-95c4c69783488dd7b39c4269e0d257f53a8f49dfdeb4994e6bfc8cf600ecc5d9R64 (`def version(offset: Offset): Byte = split(offset)._1`)?

Reply:
Oh, missed that! Thanks.